groth16/src/lib.rs

```diff
@@ -1,5 +1,285 @@
+pub mod export_solidity_verifier;
 pub mod groth16;
 
 pub use bellman_ce::pairing::ff;
 pub use ff::*;
 pub use franklin_crypto::bellman as bellman_ce;
+
+mod parse {
+    use crate::bellman_ce::pairing::{bls12_381::Bls12, bn256::Bn256};
+    use algebraic::{PrimeField, PrimeFieldRepr};
+    use franklin_crypto::bellman::CurveAffine;
+
+    pub fn render_scalar_to_hex<F: PrimeField>(el: &F) -> String {
+        let mut buff = vec![];
+        let repr = el.into_repr();
+        repr.write_be(&mut buff).unwrap();
+
+        format!("0x{}", hex::encode(buff))
+    }
+
+    fn parse_g1(
+        e: &<Bn256 as franklin_crypto::bellman::pairing::Engine>::G1Affine,
+    ) -> (String, String) {
+        let (x, y) = e.into_xy_unchecked();
+
+        (render_scalar_to_hex(&x), render_scalar_to_hex(&y))
+    }
+
+    fn parse_g2(
+        e: &<Bn256 as franklin_crypto::bellman::pairing::Engine>::G2Affine,
+    ) -> (String, String, String, String) {
+        let (x, y) = e.into_xy_unchecked();
+        (
+            render_scalar_to_hex(&x.c0),
+            render_scalar_to_hex(&x.c1),
+            render_scalar_to_hex(&y.c0),
+            render_scalar_to_hex(&y.c1),
+        )
+    }
+
+    pub fn parse_g1_json(
+        e: &<Bn256 as franklin_crypto::bellman::pairing::Engine>::G1Affine,
+    ) -> String {
+        let parsed = parse_g1(e);
+
+        format!("[\"{}\", \"{}\"]", parsed.0, parsed.1)
+    }
+
+    pub fn parse_g2_json(
+        e: &<Bn256 as franklin_crypto::bellman::pairing::Engine>::G2Affine,
+    ) -> String {
+        let parsed = parse_g2(e);
+
+        format!(
+            "[[\"{}\", \"{}\"], [\"{}\", \"{}\"]]",
+            parsed.0, parsed.1, parsed.2, parsed.3,
+        )
+    }
+
+    fn parse_g1_bls12381(
+        e: &<Bls12 as franklin_crypto::bellman::pairing::Engine>::G1Affine,
+    ) -> (String, String) {
+        let (x, y) = e.into_xy_unchecked();
+
+        (render_scalar_to_hex(&x), render_scalar_to_hex(&y))
+    }
+
+    fn parse_g2_bls12381(
```

Comment:
fn parse_g2<E: franklin_crypto::bellman::pairing::Engine>(e: E::G2Affine) ->  (String, String, String, String) {
      let (x, y) = e.into_xy_unchecked();
        (
            render_scalar_to_hex(&x.c0),
            render_scalar_to_hex(&x.c1),
            render_scalar_to_hex(&y.c0),
            render_scalar_to_hex(&y.c1),
        )
}

Then you can call by  parse_g2::<bls12381>(xxx).